src/engagement_db_to_analysis/rapid_pro_advert_functions.py

```diff
@@ -209,25 +210,28 @@ def _sync_group_to_rapid_pro(cache, target_uuids, group_name, uuid_table, rapid_
 
     synced_uuids = []
     if cache is not None:
-        synced_dataset_nc_uuids = cache.get_synced_uuids(group_name)
-        log.info(f'Found {len(synced_dataset_nc_uuids)} previously uploaded to {group_name}...')
+        previously_synced_non_relevant_uuids = cache.get_synced_uuids(advert_contact_field_name)
+        log.info(f'Found {len(previously_synced_non_relevant_uuids)} uuids whose {advert_contact_field_name} contact '
+                 f'field was synced in previous pipeline run...')
+        synced_uuids.extend(previously_synced_non_relevant_uuids)
 
     # If cache is available, check for uuids to sync in the current pipeline run.
     uuids_to_sync = _get_uuids_to_sync(target_uuids, synced_uuids)
 
-    # Re-identify the uuids.
-    urns_to_sync = _convert_uuids_to_urns(uuids_to_sync, uuid_table)
+    if len(uuids_to_sync) > 0:
+        log.info(f'Syncing {len(uuids_to_sync)} urns in this run ')
+        # Re-identify the uuids.
+        urns_to_sync = _convert_uuids_to_urns(uuids_to_sync, uuid_table)
+        # Update the advert contact field for the target urns.
+        for urn in urns_to_sync:
+            rapid_pro.update_contact(urn, contact_fields={advert_contact_field_name: "yes"})
+            synced_uuids.append(uuid_table.data_to_uuid(urn))
 
-    # Sync the group contacts to rapid_pro.
-    log.info(f'Adding {len(urns_to_sync)} contacts to {group_name} group...')
-    advert_group_uuid = _ensure_rapid_pro_group_exists(group_name, rapid_pro)
+        if cache is not None:
+            cache.set_synced_uuids(advert_contact_field_name, synced_uuids)
 
-    for urn in urns_to_sync:
-        _add_urn_to_group(urn, advert_group_uuid, rapid_pro)
-        synced_uuids.append(uuid_table.data_to_uuid(urn))
-
-    if cache is not None:
-        cache.set_synced_uuids(group_name, synced_uuids)
+    else:
+        log.info(f'Found {len(uuids_to_sync)} uuids to sync in this run skipping...')
```

Comment:
I think this reads better as
```suggestion
        log.info(f'Found 0 uuids to sync in this run skipping...')
```
if you're using len(uuids_to_sync) as a check I think it's better to put that check in an assert